libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -4642,10 +4648,10 @@ void GCS_MAVLINK::send_sys_status()
 
     const AP_BattMonitor &battery = AP::battery();
     float battery_current;
-    uint8_t battery_remaining = -1;
+    int8_t battery_remaining = -1;
```

Comment:
I think you no longer need to initialize this now.

```suggestion
    int8_t battery_remaining;
```

Reply:
We still need it I guess. `battery_remaining` is assigned the value only if the battery is healthy. So somehow if we fail to get the percentage, we send -1 with `battery_remaining`.

Reply:
In the strictest sense sure if someone added a bug to your function `GCS_MAVLINK::battery_remaining_pct()`
If, `AP::battery().capacity_remaining_pct(percentage) == false` : Your function returns -1
However, IMO it is just redundant for this case.

Now there might some future, not this PR, work to figure if the battery is unhealthy and return false for other cases in 
`AP_BattMonitor_Backend::capacity_remaining_pct()` other than `_params._pack_capacity <= 10`

Reply:
Yes. I am also just worried about `battery.healty()`, few lines below we initialise `battery_remaining`. I'll try to think what we can do to handle this in `battery_remaining_pct()` itself. Thanks for taking your time to review my PR.

Reply:
Opps sorry ignore me I didn't scroll down low enough. You are right you would need to initialize here currently.

We would need to add some more logic to AP_BattMonitor_Backend::capacity_remaining_pct() which would make sense to me to do so but doesn't need to be in this PR.